cmd/local-admin-client.go

```diff
@@ -110,3 +115,54 @@ func (lc localAdminClient) DownloadProfilingData() ([]byte, error) {
 
 	return data, nil
 }
+
+// ServerDrivePerfInfo - Drive performance information in MB/s
+func (lc localAdminClient) ServerDrivePerfInfo() (drivePerf []DriveWriteSpeed, e error) {
+	if globalBootTime.IsZero() {
+		return drivePerf, errServerNotInitialized
+	}
+
+	var drivesInfo []DriveWriteSpeed
+	// Loop through all the end points
+	for _, endpoint := range globalEndpoints {
+		// Only use local endpoints here
+		if endpoint.IsLocal {
+			if _, err := os.Stat(endpoint.Path); err != nil {
+				// Since this drive/endpoint is not available, continue
+				continue
+			}
+			var writeCount int
+			var totalElapsed time.Duration
+			var driveInfo DriveWriteSpeed
+			// Create a file 10 times in series, then find the average write time
+			for writeCount = 1; writeCount <= 10; writeCount++ {
+				buf := make([]byte, readSizeV1)
+				// Randomize values in the buffer
+				rand.Read(buf)
+				data := bytes.NewReader(buf)
+				tempObj := mustGetUUID()
+				fsTmpObjPath := pathJoin(endpoint.Path, minioMetaTmpBucket, tempObj)
+
+				start := time.Now()
+				_, err := fsCreateFile(context.Background(), fsTmpObjPath, data, nil, data.Size())
```

Comment:
Do not use fsCreateFile, it has overahead create custom functions for this which are well suited for performance needs.